### PR TITLE
Receive more messages from mfe

### PR DESCRIPTION
**Pull request by ndricimrr**
**Description**

Changes proposed in this pull request:

- Add more message types
- Add examples to test
- 

**How to test/try out:**
- cd container
- npm run start
- Go to http://localhost:8080/index2.html 
- Click on the buttons on the shown microfrontend to trigger the client calls and check that the messages are received properly

### CODE REVIEW

File: container/src/services/container.service.ts [hunk 1]
```diff
@@ -43,24 +46,54 @@ export class ContainerService {
...
+                this.dispatch(Events.ALERT_REQUEST, targetCnt, event.data.params);
                 break;
-              case 'luigi.ux.alert.show':
-                this.dispatch('alert-request', targetCnt, event.data.params);
+              case LuigiMessageID.INITIALIZED:
+                this.dispatch(Events.INITIALIZED, targetCnt, event.data.params);
                 break;
-              case 'luigi.init.ok':
-                this.dispatch('initialized', targetCnt, event.data.params);
+
```
Comment (ndricimrr) [inline container/src/services/container.service.ts:70]:
```suggestion
```

File: container/src/services/container.service.ts [hunk 2]
```diff
@@ -43,24 +46,54 @@ export class ContainerService {
...
+
+              // what it is
```
Comment (ndricimrr) [inline container/src/services/container.service.ts:72]:
```suggestion
```

File: container/src/services/container.service.ts [hunk 3]
```diff
@@ -43,24 +46,54 @@ export class ContainerService {
...
+              case LuigiMessageID.ADD_SEARCH_PARAMS_REQUEST:
+                this.dispatch(Events.ADD_SEARCH_PARAMS_REQUEST, targetCnt, { data: event.data.data, keepBrowserHistory: event.data.keepBrowserHistory });
                 break;
+
+              // what it is
```
Comment (ndricimrr) [inline container/src/services/container.service.ts:77]:
```suggestion
```

File: container/src/services/container.service.ts [hunk 4]
```diff
@@ -43,24 +46,54 @@ export class ContainerService {
...
+              case LuigiMessageID.ADD_NODE_PARAMS_REQUEST:
+                this.dispatch(Events.ADD_NODE_PARAMS_REQUEST, targetCnt, { data: event.data.data, keepBrowserHistory: event.data.keepBrowserHistory });
+                break;
+
+              // what it is
```
Comment (ndricimrr) [inline container/src/services/container.service.ts:82]:
```suggestion
```

File: container/src/services/container.service.ts [hunk 5]
```diff
@@ -43,24 +46,54 @@ export class ContainerService {
...
+              case LuigiMessageID.SHOW_CONFIRMATION_MODAL_REQUEST:
+                this.dispatch(Events.SHOW_CONFIRMATION_MODAL_REQUEST, targetCnt, event.data.data);
+                break;
+
+              // what it is
+              case LuigiMessageID.SHOW_LOADING_INDICATOR_REQUEST:
+                this.dispatch(Events.SHOW_LOADING_INDICATOR_REQUEST, targetCnt, event);
+                break;
+
+              // what it is
```
Comment (ndricimrr) [inline container/src/services/container.service.ts:92]:
```suggestion
```

File: container/src/services/container.service.ts [hunk 6]
```diff
@@ -43,24 +46,48 @@ export class ContainerService {
...
+              case LuigiMessageID.NAVIGATION_REQUEST:
+                this.dispatch(Events.NAVIGATION_REQUEST, targetCnt, event.data.params);
+                break;
+              case LuigiMessageID.ALERT_REQUEST:
+                this.dispatch(Events.ALERT_REQUEST, targetCnt, event.data.params);
+                break;
+              case LuigiMessageID.INITIALIZED:
+                this.dispatch(Events.INITIALIZED, targetCnt, event.data.params);
+                break;
+
```
Comment (ndricimrr) [inline container/src/services/container.service.ts:70]:
```suggestion

```

File: container/src/services/container.service.ts [hunk 7]
```diff
@@ -43,24 +46,48 @@ export class ContainerService {
...
+              case LuigiMessageID.ADD_SEARCH_PARAMS_REQUEST:
+                this.dispatch(Events.ADD_SEARCH_PARAMS_REQUEST, targetCnt, { data: event.data.data, keepBrowserHistory: event.data.keepBrowserHistory });
                 break;
-              case 'luigi.get-context':
-                target.postMessage({ msg: 'luigi.init', context: targetCnt.context || {}, internal: {} }, '*');
+
```
Comment (ndricimrr) [inline container/src/services/container.service.ts:74]:
```suggestion

```

File: container/src/services/container.service.ts [hunk 8]
```diff
@@ -43,24 +46,48 @@ export class ContainerService {
...
+              case LuigiMessageID.ADD_NODE_PARAMS_REQUEST:
+                this.dispatch(Events.ADD_NODE_PARAMS_REQUEST, targetCnt, { data: event.data.data, keepBrowserHistory: event.data.keepBrowserHistory });
                 break;
-              case 'luigi.navigation.open':
-                this.dispatch('navigation-request', targetCnt, event.data.params);
+
```
Comment (ndricimrr) [inline container/src/services/container.service.ts:78]:
```suggestion

```

File: container/src/services/container.service.ts [hunk 9]
```diff
@@ -43,24 +46,48 @@ export class ContainerService {
...
+              case LuigiMessageID.SHOW_CONFIRMATION_MODAL_REQUEST:
+                this.dispatch(Events.SHOW_CONFIRMATION_MODAL_REQUEST, targetCnt, event.data.data);
                 break;
-              case 'luigi.ux.alert.show':
-                this.dispatch('alert-request', targetCnt, event.data.params);
+
```
Comment (ndricimrr) [inline container/src/services/container.service.ts:82]:
```suggestion

```

File: container/src/services/container.service.ts [hunk 10]
```diff
@@ -43,24 +46,48 @@ export class ContainerService {
...
+              case LuigiMessageID.SHOW_LOADING_INDICATOR_REQUEST:
+                this.dispatch(Events.SHOW_LOADING_INDICATOR_REQUEST, targetCnt, event);
                 break;
-              case 'luigi.init.ok':
-                this.dispatch('initialized', targetCnt, event.data.params);
+
+              case LuigiMessageID.HIDE_LOADING_INDICATOR_REQUEST:
+                this.dispatch(Events.HIDE_LOADING_INDICATOR_REQUEST, targetCnt, event);
                 break;
+
```
Comment (ndricimrr) [inline container/src/services/container.service.ts:90]:
```suggestion

```

File: container/src/services/container.service.ts [hunk 11]
```diff
@@ -43,23 +46,41 @@ export class ContainerService {
...
           if (target === event.source && (event.data?.msg?.indexOf('luigi.') === 0 || event.data?.msg === 'custom')) {
+            // messages emitted from microfrontends
             const msg = event.data.msg;
 
+            // dispatch an event depending on message
             switch (msg) {
-              case 'custom':
-                this.dispatch('custom-message', targetCnt, event.data.data);
+              case LuigiMessageID.CUSTOM_MESSAGE:
+                console.log('Test', targetCnt, event)
```
Comment (ndricimrr) [inline container/src/services/container.service.ts:55]:
```suggestion

```

File: container/src/LuigiContainer.svelte [hunk 12]
```diff
@@ -65,18 +65,37 @@
     deferInit = false;
   };
 
+  thisComponent.addAlertListener = (domManipulationFn, data) => {
```
Comment (ndricimrr) [inline container/src/LuigiContainer.svelte:68]: todo: remove and wrap in another file maybe
todo: cleanup

File: container/src/constants/communication.ts [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+
+    /**
+     * A message emmitted from the Microfrontend when a request to receive the current app route is sent
+     */
+    export const GET_CURRENT_ROUTE_REQUEST = 'get-current-route-request';
+
+    /**
+     * A message emmitted from the CORE ? MICROFRONTEND ?????????? when a request to report the issued token to client is sent
+     */
+    export const REPORT_TOKEN_ISSUED_REQUEST = 'report-token-issued-request';
```
Comment (ndricimrr) [inline container/src/constants/communication.ts:83]: remove

File: container/src/services/container.service.ts [hunk 14]
```diff
@@ -43,28 +48,82 @@ export class ContainerService {
...
+                this.dispatch(Events.NAVIGATION_COMPLETED_REPORT, targetCnt, event);
+                break;
+              case LuigiMessageID.UPDATE_MODAL_PATH_DATA_REQUEST:
+                this.dispatch(Events.UPDATE_MODAL_PATH_DATA_REQUEST, targetCnt, event);
+                break;
+              case LuigiMessageID.CHECK_PATH_EXISTS_REQUEST:
+                this.dispatch(Events.CHECK_PATH_EXISTS_REQUEST, targetCnt, event);
+                break;
+              case LuigiMessageID.SET_DIRTY_STATUS_REQUEST:
+                this.dispatch(Events.SET_DIRTY_STATUS_REQUEST, targetCnt, event);
```
Comment (ndricimrr) [inline container/src/services/container.service.ts:122]: expand with the rest of the message ids

File: container/src/services/container.service.ts [hunk 15]
```diff
@@ -43,28 +48,82 @@ export class ContainerService {
...
+                this.dispatch(Events.GET_CONTEXT_REQUEST, targetCnt, event.data, (data: any) => {
+                  console.log('Callback called: Received data from Core sending inside MF', data);
+                  target.postMessage({ msg: LuigiMessageID.SEND_CONTEXT, context: data }, '*')
+                }, LuigiCoreApi.SEND_CONTEXT_TO_MICROFRONTEND);
+                break;
+              case LuigiMessageID.NAVIGATION_REQUEST:
+                this.dispatch(Events.NAVIGATION_REQUEST, targetCnt, event.data.params);
+                break;
+              case LuigiMessageID.ALERT_REQUEST:
+                this.dispatch(Events.ALERT_REQUEST, targetCnt, event.data.params);
```
Comment (ndricimrr) [inline container/src/services/container.service.ts:71]: handle different alert ids

**Review by JohannesDoberer — APPROVED** [review]
LGTM 👍 !

**Comment by ndricimrr** [comment]
add release new rc

**Pull request merged**